src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs

```diff
@@ -845,6 +863,7 @@ private void Dispose(bool disposing)
 
         private void EnableReceive()
         {
+            Debug.Assert(!Monitor.IsEntered(_state), "!Monitor.IsEntered(_state)");
```

Comment:
Instead of repeating all this, it might be a little more readable to just call a method?

```c#
[Conditional("DEBUG")]
internal void AssertMonitorNotEntered(object obj)
{
     Debug.Assert(!Monitor.IsEntered(obj), "Monitor was unexpectedly held");
}
```

Reply:
(If we do https://github.com/dotnet/runtime/issues/65965, it'll also be redundant in this case, as with or without the message you'd get the same assert.)